core/app/models/spree/promotion/rules/nth_order.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  class Promotion
+    module Rules
+      class NthOrder < PromotionRule
+        preference :nth_order, :integer, default: 2
+        # It does not make sense to have this apply to the first order using preferred_nth_order == 1
+        # Instead we could use the first_order rule
```

Comment:
It does require a user already, but what I'm trying to say is it that it seems like a hard thing, for example, to message to a random store using solidus -- "If you use NthOrder make sure you understand xxx because it behaves unexpectedly if you don't require login". Maybe we could have an application configuration that allows NthOrder to work for n=1?  Then people wouldn't accidentally get unexpected behavior. Are you hoping to reduce the number of promo types for our admins since NthOrder n=1 works in our case?